src/lib/y2storage/encryption_processes/base.rb

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Storage
+  module EncryptionProcesses
+    # Base class to perform the creation of an encrypted device
+    #
+    # @note This is a base class. To really created an encrypted device use any
+    # of its subclasses, which must define an #enccryption_type
+    class Base
+      include Yast::Logger
+
+      # Whether the process was used for the given encryption device
+      #
+      # @param _encryption [Y2Storage::Encryption] the encryption device to check
+      # @return [Boolean] true if the given device looks to be encrypted with
+      # the process; false otherwise
+      def self.used_for?(_encryption)
+        false
+      end
+
+      # Constructor
+      #
+      # @param method [Y2Storage::EncryptionMethod]
+      def initialize(method)
+        @method = method
+      end
+
+      attr_reader :method
+
+      # Creates and returns an encryption device with given params
+      #
+      # @param blk_device [Y2Storage::BlkDevice]
+      # @param dm_name [String]
+      def create_device(blk_device, dm_name)
+        enc = blk_device.create_encryption(dm_name || "", encryption_type)
+        enc.encryption_process = self
+        enc
+      end
+
+      private
+
+      # Returns the Y2Storage::EncryptionType to be used
```

Comment:
NP: I would not mention the real type in the description text: Returns the encryption type to be used. The real type is indicated with the `@return` tag.

Reply:
BTW, we are using `abstract_method` for these "abstracts methods" (actually, ruby lacks of the abstract method concept): https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2partitioner/actions/base.rb#L58

Reply:
Thank you!